Slim/Music/Info.pm

```diff
@@ -476,7 +476,8 @@ sub setRemoteMetadata {
 		$currentBitrates{$url} = $track->prettyBitRate;
 	}
 	
-	if ( $meta->{cover} && $url =~ m|^http| ) {
+	if ( $meta->{cover} && $url =~ m|https?://| ) {
+		# cache only if it's HTTP(s) - direct or hidden
```

Comment:
What do you consider a "hidden" URL? When would we have a url which hadn't the `http` at the beginning?

Reply:
Yes, something like myph://http(s)://. I agree I could make the regex more specific

Reply:
Instead of somehow make this work here, would it make sense to explicitly cache the artwork in the PH instead?

Reply:
That was one of the biggest hurdles: if you look at Slim::Player::Protocols::HTTP::getMetadataFor it does the same. When you add a track to the playlist, there is no other way I can think of to grab the image that was used. 
I mean, you have the OPML list with a series of items in  menu with an îtem->{image} entry. When user adds one of these to the active playlist, all you get is the $url that was in the menu and then a call to getMetadataFor. So, in LMS currently, the XMLBrowser caches the remote image so that the getMetadataFor can grab it, "on the other side of the pond".
The problem is that if XMLBrowser does not cache it, then I can't grab it

Reply:
ok

Reply:
I have to come back to this, sorry. You'd only ever have a hidden URL when using a custom protocol handler. And any URL coming with a custom protocol has gone through some kind of URL mangling - the `wrap()` function in the case of the podcast. Therefore we could have that wrapper do the caching with the new URL, then extend `getMetadataForUrl()` in the protocol handler. In the simplest case we'd call the parent's method and only handle the artwork in that PH local implementation. What do you think? I believe that's pretty much how I do it in some of my plugins. This would also solve the issue of URI escaping the URL.

Reply:
I'm not sure I understand how to do that. I mean that the getMetadata of the PH would have no way to know where to get the "image" for that "url" unless in the wrapper, you cache *every* entry of the OPML list vs today only the ones that are being added to the actual playlist. That's a lot of inflation of an already busy cache, no? 

Still, I thought about doing that earlier but I decided against for an additional reason which is you have to follow redirection because at some point the PH will call you with the redirected "url" and unless somebody like S::U::S::Remote::redirect does it for you, you'll get lost. In other PH, you get the metadata through a different way, but not through scanURL which *does* change the track's url. So with other custom PH, your $track->url is constant but not with scanUrl. We lose track of things when using scanURL. Not using scanURL is a *major* change.